update_prevalence.py

```diff
@@ -235,6 +288,66 @@ def cases_week_before(self) -> int:
         ...
 
+    def set_total_vaccines(self, partial_vaccinations: int, complete_vaccinations: int):
+        self.vaccines_total.partial_vaccinations = partial_vaccinations
+        self.vaccines_total.completed_vaccinations = complete_vaccinations
+    
+    def set_vaccines_of_type(self, vaccine_type: str, partial: int, complete: int):
+        if self.vaccines_by_type is None:
+            self.vaccines_by_type = {}
+        
+        self.vaccines_by_type[vaccine_type] = Vaccination()
+        self.vaccines_by_type[vaccine_type].partial_vaccinations = partial
+        self.vaccines_by_type[vaccine_type].completed_vaccinations = complete
+
+    def completed_vaccination_total(self):
+        if (self.vaccines_by_type is not None):
+            return reduce(lambda x, key: x + self.vaccines_by_type[key].completed_vaccinations, self.vaccines_by_type, 0)
+        return self.vaccines_total.completed_vaccinations
+
+    def partial_vaccination_total(self):
+        if (self.vaccines_by_type is not None):
+            return reduce(lambda x, key: x + self.vaccines_by_type[key].partial_vaccinations, self.vaccines_by_type, 0)
+        return self.vaccines_total.partial_vaccinations
+
+    # Compute the estimated risk ratio for an unvaccinated person vs an average person.
+    # Average prevalence = sum(prevalence in group * proportion of population in group)
+    #                    = sum(vaccine_mult * unvaccinated_prev * proportion of population)
+    # Unvaccinated risk / Average risk = sum(vaccine_mult * proportion of pop)
+    #                                  = population / sum(vaccine_mult * # of vaccinations)
+    def unvaccinated_relative_prevalence(self) -> float:
+        total_vaccinated = 0
+        risk_sum = 0
+
+        if (self.vaccines_by_type is not None):
+            for vaccine_type, vaccine_status in self.vaccines_by_type.items():
+                total_vaccinated += vaccine_status.completed_vaccinations
+                total_vaccinated += vaccine_status.partial_vaccinations
+                risk_sum += VACCINE_MULTIPLIERS[vaccine_type]['complete'] * vaccine_status.completed_vaccinations
+                risk_sum += VACCINE_MULTIPLIERS[vaccine_type]['partial'] * vaccine_status.partial_vaccinations
+        else:
+            risk_sum = (
+                VACCINE_MULTIPLIERS['Unknown']['complete'] * 
+                self.vaccines_total.completed_vaccinations +
+                VACCINE_MULTIPLIERS['Unknown']['complete'] * 
+                self.vaccines_total.partial_vaccinations
+            )
+            total_vaccinated = (
+                self.vaccines_total.completed_vaccinations + 
+                self.vaccines_total.partial_vaccinations
+            )
+
+        assert (
+            total_vaccinated <= self.population
+        ), f"{self.label} has over 100% vaccinated. Population: {self.population}, vaccinations: {total_vaccinated}"
+
+        if total_vaccinated == 0:
+            return 
```

Comment:
Shouldn't this case just return 1.0?

Reply:
I interpret 0 vaccines as not having data, which means we should refuse to compute "average vaccinated" This will propagate through as a null field in the data export.